open_feature/hooks/hook_support.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import typing
+from functools import reduce
+
+from open_feature.evaluation_context.evaluation_context import EvaluationContext
+from open_feature.flag_evaluation.flag_evaluation_details import FlagEvaluationDetails
+from open_feature.flag_evaluation.flag_type import FlagType
+from open_feature.hooks.hook import Hook
+from open_feature.hooks.hook_context import HookContext
+
+
+def error_hooks(
+    flag_type: FlagType,
+    hook_context: HookContext,
+    exception: Exception,
+    hooks: typing.List[Hook],
+    hints: dict,
+):
+    kwargs = {"ctx": hook_context, "exception": exception, "hints": hints}
+    execute_hooks(flag_type=flag_type, hooks=hooks, hook_method="error", **kwargs)
+
+
+def after_all_hooks(
+    flag_type: FlagType, hook_context: HookContext, hooks: list, hints: dict
+):
+    kwargs = {"ctx": hook_context, "hints": hints}
+    execute_hooks(
+        flag_type=flag_type, hooks=hooks, hook_method="finally_after", **kwargs
+    )
+
+
+def after_hooks(
+    flag_type: FlagType,
+    hook_context: HookContext,
+    details: FlagEvaluationDetails,
+    hooks: typing.List[Hook],
+    hints: dict,
+):
+    kwargs = {"ctx": hook_context, "details": details, "hints": hints}
+    execute_hooks_unchecked(
+        flag_type=flag_type, hooks=hooks, hook_method="after", **kwargs
+    )
+
+
+def before_hooks(
+    flag_type: FlagType,
+    hook_context: HookContext,
+    hooks: typing.List[Hook],
+    hints: dict,
+) -> EvaluationContext:
+    kwargs = {"ctx": hook_context, "hints": hints}
+    executed_hooks = execute_hooks(
+        flag_type=flag_type, hooks=hooks, hook_method="before", **kwargs
+    )
+    filtered_hooks = list(filter(lambda hook: hook is None, executed_hooks))
+
+    if filtered_hooks:
+        return reduce(lambda a, b: a.merge(b), filtered_hooks)
+    else:
+        return EvaluationContext()
+
+
+def execute_hooks(
```

Comment:
I think this is probably only called by the functions defined above so could this be marked as a private function? `_execute_hooks()` (question also applies to the 2 functions below)